system/osx_defaults.py

```diff
@@ -160,6 +172,15 @@ def _convert_type(self, type, value):
             if not isinstance(value, list):
                 raise OSXDefaultsException("Invalid value. Expected value to be an array")
             return value
+        elif type in ["dict", "dictionary"]:
+            if isinstance(value, dict):
+                return [value]
+            elif isinstance(value, list):
+                return value
+            elif isinstance(value, str):
```

Comment:
/cc @sivel 

```
sizeup_defaults:
  macosx:
    - { key: CenterRelativeResizeEnabled, value: 1, type: bool }
    - { key: CenterResizeType, value: 1, type: int }
    - { key: CenterRelativeResizeHeight, value: 85, type: int }
    - { key: CenterRelativeResizeWidth, value: 85, type: int }
    - { key: ShowOverlay, value: 0, type: bool }
    - { key: ShowTooltips, value: 0, type: bool }
    - { key: Lower Left, type: dict, value: {key: ComboFlags, type: int, value: 10092544} }
    - { key: Lower Left, type: dict, value: {key: ComboCode, type: int, value: 125} }
    - { key: Lower Right, type: dict, value: {key: ComboFlags, type: int, value: 10092544} }
    - { key: Lower Right, type: dict, value: {key: ComboCode, type: int, value: 124} }
    - { key: Upper Left, type: dict, value: {key: ComboFlags, type: int, value: 10092544} }
    - { key: Upper Left, type: dict, value: {key: ComboCode, type: int, value: 123} }
    - { key: Upper Right, type: dict, value: {key: ComboFlags, type: int, value: 10092544} }
    - { key: Upper Right, type: dict, value: {key: ComboCode, type: int, value: 126} }
```

In this case, it's the value: {} coming through as a string, even through type: dict exists. Why I don't yet get is why this happens in 2.1, but not 2.0

The playbook bits:

```
- name: Configure SizeUp Defaults
  osx_defaults:
    domain: com.irradiatedsoftware.SizeUp
    key: '{{ item.key }}'
    type: '{{ item.type }}'
    value: '{{ item.value }}'
    state: present
    dict_add: True
  with_items: '{{ sizeup_defaults.macosx }}'
  when: ansible_distribution == 'MacOSX'
  tags: sizeup-config
```